src/ingredient.rs

```diff
@@ -38,7 +39,7 @@ pub trait Ingredient: Any + std::fmt::Debug + Send + Sync {
         db: &'db dyn Database,
         input: Option<Id>,
         revision: Revision,
-    ) -> bool;
+    ) -> VerifyResult;
```

Comment:
I think this change will also come handy for the "faster accumulator" work because we'll need to also return whether the ingredient had any accumulated values.